### ISSUE TITLE
Changed sequelize.or behavior

### ISSUE THREAD

**Issue by BridgeAR** [sequelize/sequelize#3107]
The refactored where logic also changed the behavior of sequelize.and and sequelize.or.

I used the following statement:
```javascript
sequelize.or({
    roleName: rolesNew
}, {
    roleName: 'CLIENT',
    type: clientType
})
```
It was translated to 
roleName: rolesNew OR roleName: 'CLIENT' AND type: clientType
after the refactor it's
roleName: rolesNew OR roleName: 'CLIENT' OR type: clientType

Of course the following works as expected:
```javascript
sequelize.or({
    roleName: rolesNew
}, sequelize.and({
    roleName: 'CLIENT'
}, {
    type: clientType
}))
```
And this works too:
```javascript
sequelize.or({
    roleName: rolesNew
}, sequelize.and({
    roleName: 'CLIENT',
    type: clientType
}))
```
When reading the doku it's not really clear if the new behavior is the more favored one or not.

**Comment by janmeier** [comment]
That behaviour is definitely not desired

**Comment by mickhansen** [comment]
Hmm no that's definitely a regression, strange it wasn't caught by a test.

**Comment by mickhansen** [comment]
It's a bit tricky, since we support stuff like:

```js
$or: {
  email: 'heyo',
  name: 'heyo'
}
```

Now, which results in an or.
So i suppose the rule is that if $or is an array then its OR between the arrays but AND inside.

**Comment by mickhansen** [comment]
Fixed in `sequelize@2.0.2`
